webserver.cpp

```diff
@@ -473,7 +475,7 @@ static int webserver_parse_post(struct webserver_t *client, uint16_t size) {
 int8_t http_parse_request(struct webserver_t *client, uint8_t **buf, uint16_t *len) {
   uint16_t hasread = MIN(WEBSERVER_BUFFER_SIZE-client->ptr, *len);
 
-  while(*len > 0) {
+  while((*len > 0) || (strnstr(client->buffer, "\r\n\r\n", client->ptr) != NULL)) {
```

Comment:
This is also necessary when the whole body fits into the client buffer. Len could already be 0 but there can still be useful data in the client buffer which needs to be parsed by all the if statements below. So, if the client buffer contains the double CRLF (which indicates end of the headers) it is safe to keep parsing the rest of the headers.